extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/filter/AutoAddOpenApiEndpointFilter.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.smallrye.openapi.deployment.filter;
+
+import org.eclipse.microprofile.openapi.OASFactory;
+import org.eclipse.microprofile.openapi.OASFilter;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.Paths;
+import org.eclipse.microprofile.openapi.models.media.Content;
+import org.eclipse.microprofile.openapi.models.responses.APIResponses;
+
+public class AutoAddOpenApiEndpointFilter implements OASFilter {
+    private static final String OPENAPI_TAG = "openapi";
+    private static final String ENDPOINT_DESCRIPTION = "Open API specification";
+
+    private final String path;
+
+    public AutoAddOpenApiEndpointFilter(String path) {
+        this.path = path;
```

Comment:
There are actually 4 paths supported:
- `/openapi.json` always returns JSON
- `/openapi.yaml` always returns YAML
- `/openapi.yml` always returns YAML
- `/openapi` returns YAML or JSON depending on `Accept` header, `format` query param, or defaults to YAML.